keps/sig-apps/3329-retriable-and-non-retriable-failures/README.md

```diff
@@ -727,6 +798,16 @@ condition makes it easier to determine if a failed pod should be restarted):
 - DeletionByTaintManager (Pod evicted by kube-controller-manager due to taints)
 - EvictionByEvictionAPI (Pod deleted by Eviction API)
 - DeletionByPodGC (an orphaned Pod deleted by pod GC)
+- DeletionByKubelet (Pod deleted due to node shutdown, node pressure or Pod admission errors)
+
+We introduce a Pod condition type, called `ResourceLimitExceeded`, used by Kubelet
+to indicate that a Pod failure is caused by exceeding pod's (or container's) resource
+limits. We add the following reason types to indicate the resource limit which
+was exceeded:
+- MemoryLimitExceeded (limit on container's memory is exceeded)
+- ContainerEphemeralStorageLimitExceeded (limit on container's emphemeral-storage is exceeded)
+- EmptyDirLimitExceeded (limit on pod's empty-dir storage is exceeded)
+- PodEphemeralStorageLimitExceeded (limit on total pod's emphemeral-storage is exceeded).
```

Comment:
Since users can't define policies in the proposed API based on those reasons, please clarify why we are defining those constants and how we expect users/controllers to make use of them.

Reply:
`We list the reason constants above in order to gain the community consensus on informative names.` as in the added paragraph. However, I would also be ok with either removing the proposed names from KEP (to leave them as an implementation detail as for messages) or with implementing support for matching by the `reason` field.

Reply:
We are just trying to be informative. However, I would just have two reasons: MemoryLimitExceeded and EphemeralStorageLimitExceeded. Otherwise, can you clarify what's the difference?

Reply:
I squeezed them into `EphemeralStorageLimitExceeded` for now. Alternatively, we could name it `LocalStorageLimitExceeded` which is the name used in the Kubelet code (https://github.com/kubernetes/kubernetes/blob/9e65786254c438942ca45d2c447d6318417fc287/pkg/kubelet/eviction/eviction_manager.go#L455) to also include `emptyDir` (which is set by the `sizeLimit` property: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.24/#emptydirvolumesource-v1-core). On the other hand `emptyDir` is also an ephemeral volume https://kubernetes.io/docs/concepts/storage/ephemeral-volumes/, so the name `EphemeralStorageLimitExceeded` sounds good.